sys/Plugins/Transports/FtpTransport.php

```diff
@@ -131,9 +141,25 @@ public function get(): array {
 
 		//ftp_chdir($conn, $this->path);
 
-		$contents = ftp_nlist($conn, $this->path);
+		$contentsAll = ftp_mlsd($conn, $this->path);
+		//die(var_dump($contentsAll));
+
+		// Take only newest of all files which meet the filename pattern
+		$path = $this->path;
+		if ($path !== '' && substr($path, -1) !== '/') $path .= '/';
+		$contents = [];
+		$newest = '0';
+		foreach ($contentsAll as $entry) {
+			if ($entry['type'] !== 'file') continue;
+			if ($this->filenamePattern !== '' && !fnmatch($this->filenamePattern, $entry['name'])) continue;
+			if (count($contents) > 0 && $entry['modify'] < $newest) continue;
+			if (count($contents) == 0) array_push($contents, $path . $entry['name']);
+			else $contents[0] = $path . $entry['name'];
```

Comment:
No need to check if the array is empty, you can just push to it.

Reply:
the array should contain only 1 element. So if it already has one, it must be overwritten.